### PR TITLE
External storage design fixes and update

### PR DESCRIPTION
**Pull request by skjnldsv**
| Before | After |
|:---:|:---:|
|capture d ecran_2018-02-22_12-17-17|capture d ecran_2018-02-22_15-41-24|
|kazam_screenshot_00004|kazam_screenshot_00002|
|capture d ecran_2018-02-22_13-16-00|capture d ecran_2018-02-22_13-16-53|

- [x] Remove the dropdown class and use a popovermenu
- [x] Add select possibility design for popovermenu
- [x] Change error/save feedback as it is not very comprehensible right now

### CODE REVIEW

File: core/css/apps.scss [hunk 1]
```diff
@@ -683,10 +683,9 @@ kbd {
 
 /* DROPDOWN ----------------------------------------------------------------- */
 .dropdown {
-	background: nc-darken($color-main-background, 8%);
-	border-bottom-left-radius: 5px;
-	border-bottom-right-radius: 5px;
-	box-shadow: 0 1px 1px $color-box-shadow;
+	background: $color-main-background;
```
Comment (skjnldsv) [inline core/css/apps.scss]: Hum, this should not have been pushed.

**Comment by skjnldsv** [comment]
Ready for reviews! @nextcloud/designers 
@MorrisJobke can you add the people in charge of the ext storage as reviewers please? :)

File: apps/files_external/js/settings.js [hunk 2]
```diff
@@ -12,31 +12,42 @@
 
 // TODO: move to a separate file
 var MOUNT_OPTIONS_DROPDOWN_TEMPLATE =
-	'<div class="drop dropdown mountOptionsDropdown">' +
-	// FIXME: options are hard-coded for now
```
Comment (pixelipo) [inline apps/files_external/js/settings.js]: IMO this comment has not been addressed in this PR - options are still hard-coded.  I would suggest either keeping the comment or creating a generator.

File: settings/css/settings.scss [hunk 3]
```diff
@@ -1105,14 +1105,14 @@ table.grid td.date {
 
 span {
 	&.success {
-		background: #37ce02;
+		background-color: $color-success;
```
Comment (pixelipo) [inline settings/css/settings.scss]: nice catch!

**Comment by pixelipo** [comment]
I'm not sure about the new status icons - I have to say I'm quite partial to the obviousness and simplicity of the old ones :) Also, new once seem to have a non-standard size - at lease as far as their padding is concerned. 28px? 16/20 (icon size) and 44/50 (container) those are sizes we use, right?

**Comment by skjnldsv** [comment]
Yeah, I'm not really convinced either, but it's already a great improvement compared to the old ones ^^
Yes, they're all with a 28px width/height. At first I started using only the grey icons, but it was confusing (especially with the checkmark) so I decided to go for something entirely different which, at least, seems pretty clear.

What do you suggest @pixelipo ? :)

**Comment by enoch85** [comment]
Does the user get any output when hoovering on the icons on what's wrong?

**Comment by enoch85** [comment]
And regarding the icons, maybe we should use the same one as in with the Desktop Client?

**Comment by skjnldsv** [comment]
@enoch85 it depends on what the server returns. Either a popup saying what's wrong or nothing at all :/

**Review by pixelipo — DISMISSED** [review]
Padding issues with the icon buttons:
image

Row should be 50px high:
image

**Comment by skjnldsv** [comment]
@pixelipo Ahaha, I love it! When you start working on fixing a bug on something which has like 3 years outdated code, you can't get away without rewriting everything X)

At first , I was just editing the .dropdown class within the apps.scss and wanted to find where it was used in server... 2hours later, I'm rewriting the entire code of the external storage! :rofl: 
malcom

Ahaha, I'll fix the whole page, you're right, since we're here, let's leave it clean! ;)

**Comment by pixelipo** [comment]
sorry, @skjnldsv ! I know I can be such a pain in the **** sometimes...

**Comment by skjnldsv** [comment]
@pixelipo don't apologise!! I wasn't blaming you!! :hugs: 
I'll fix it after my work on the update notifications to vue.

**Comment by skjnldsv** [comment]
All clear @pixelipo

**Comment by skjnldsv** [comment]
Oups! :)

**Comment by MorrisJobke** [comment]
* [ ] I can't click the gear, trashbin or checkmark icon - there doesn't happen anything (cleared browser cache and tried Safari and Chrome)
* [ ] a little UI papercut: the checkmark has no top border:
bildschirmfoto 2018-02-27 um 11 26 28

**Comment by skjnldsv** [comment]
@MorrisJobke Indeed, it's fixed. Forgot to update a section of my code :)

**Review by MorrisJobke — APPROVED** [review]
Looks awesome and works 👍

**Comment by MorrisJobke** [comment]
JSUnit tests for files_external fail now

**Review by pixelipo — APPROVED** [review]
Looks good to me

**Pull request merged**